src/app.ts

```diff
@@ -44,13 +50,23 @@ class HyperlaneTokenApp<
 }
 
 export class HypERC20App extends HyperlaneTokenApp<HypERC20Factories> {
+  router(
+    contracts: HyperlaneContracts<HypERC20Factories>,
+  ): HypERC20 | HypERC20Collateral | HypNativeCollateral {
+    return (
+      contracts[TokenType.synthetic] ||
+      contracts[TokenType.collateral] ||
+      contracts[TokenType.native]
+    );
+  }
+
   async transfer(
     origin: ChainName,
     destination: ChainName,
     recipient: types.Address,
     amount: BigNumberish,
   ) {
-    const originRouter = this.getContracts(origin).router;
+    const originRouter = this.router(this.getContracts(origin));
```

Comment:
I would really like us to detect the contract type and:
- approve ERC20Collateral
- Set msg.value if NativeCollateral

Reply:
Probably best for a follow up PR that covers what we discussed yesterday (i.e. passing chain/tokenType to the app constructor)

Reply:
Another request I'd love to have in that PR is a `getBalance(chain)` function that will give you the balance of the native asset, collateral asset, or wrapped asset, depending on the chain